src/oc/storage/api/entries.clj

```diff
@@ -361,6 +379,69 @@
   :handle-unprocessable-entity (fn [ctx]
     (api-common/unprocessable-entity-response (:reason ctx))))
 
+;; A resource for reverting to a specific revision number.
+(defresource revert-version [conn org-slug board-slug entry-uuid]
+  (api-common/open-company-authenticated-resource config/passphrase) ; verify validity and presence of required JWToken
+
+  :allowed-methods [:options :post]
+
+  ;; Authorization
+  :allowed? (by-method {
+    :options true
+    :post (fn [ctx] (access/allow-authors conn org-slug (:user ctx)))})
+
+  ;; Media type client accepts
+  :available-media-types (by-method {
+                            :post [mt/entry-media-type]})
+  :handle-not-acceptable (by-method {
+                            :post (api-common/only-accept 406 mt/entry-media-type)})
+
+  ;; Possibly no data to handle
+  :malformed? (by-method {
+    :options false
+    :post (fn [ctx] (api-common/malformed-json? ctx true))}) ; allow nil
+  :processable? (by-method {
+    :options true
+    :post (fn [ctx] (let [entry-props (:data ctx)]
+                      (valid-entry-revert? entry-props)))})
+  :new? false
+  :respond-with-entity? true
+
+  ;; Existentialism
+  :can-post-to-missing? false
+  :exists? (fn [ctx]
+             (if-let* [_slugs? (and (slugify/valid-slug? org-slug)
+                                    (slugify/valid-slug? board-slug))
+                       org (org-res/get-org conn org-slug)
+                       org-uuid (:uuid org)
+                       entry (entry-res/get-entry conn entry-uuid)
+                       existing-version (if (zero? (:revision-id (:data ctx)))
+                                          entry
+                                          (entry-res/get-version
+                                           conn
+                                           entry-uuid
+                                           (:revision-id (:data ctx))))
+                       board (board-res/get-board conn (:board-uuid entry))
+                      _matches? (and (= org-uuid (:org-uuid entry))
+                                     (= org-uuid (:org-uuid board)))]
```

Comment:
Do we need to check if the uuids correspond here? Same as for the slugs they are checked on entry creation too.

Reply:
This is also copied but I think this is a test to see if the entry info matches the specified org still.